lib/web/app/handler.go

```diff
@@ -337,7 +337,7 @@ func HasName(r *http.Request, proxyPublicAddrs []utils.NetAddr) (string, bool) {
 	u := url.URL{
 		Scheme: "https",
 		Host:   proxyPublicAddrs[0].String(),
-		Path:   fmt.Sprintf("/web/launch/%v", raddr.Host()),
+		Path:   fmt.Sprintf("/web/launch/%v?path=%v", raddr.Host(), r.URL),
```

Comment:
Good call. I think `URL.Path` would be better suited in this case and is what I was intending to use. URL isn't always guaranteed to be just the relative path (which is was returning here rather than the full URL)

Reply:
Any query string passed would be not preserved. We could think about passing any/all arbitrary query params through the redirect process, although I've only seen asks for the URL so far. Perhaps out of scope? (but also, solving the same problem in some cases I suppose).